src/search/searchBar.component.tsx

```diff
@@ -322,14 +322,7 @@ const SearchBar = (props: SearchBarProps): React.ReactElement => {
 
     setMaxShots(maxShotsParam);
     setParamsUpdated(false);
-  }, [
-    dateRange.fromDate,
-    dateRange.toDate,
-    experimentID,
-    maxShotsParam,
-    shotnumRange.max,
-    shotnumRange.min,
-  ]);
+  }, [dateRange, experimentID, maxShotsParam, shotnumRange]);
```

Comment:
The second useEffect works better because it tracks the entire object, ensuring the effect runs whenever the object itself changes, whereas the first only tracks specific properties, potentially missing updates when the object is replaced.